File: measurecompute/DAQ/C#/ULTI01/ULTI01.cs
// ==============================================================================
//
//  File:                         ULTI01.CS
//
//  Library Call Demonstrated:    Mccdaq.MccBoard.TIn()
//
//  Purpose:                      Reads a multiplexer input channel.
//
//  Demonstration:                Displays the temperature input on a MUX
//                                channel.
//
//  Other Library Calls:          MccDaq.MccService.ErrHandling()
//
//  Special Requirements:         Unless the board at BoardNum(=0) does not use
//                                EXP boards for temperature measurements(the
//                                CIO-DAS-TC,for example), it must
//                                have an A/D converter with an attached EXP
//                                board.  Thermocouples must be wired to EXP
//                               channels selected.
//
// ==============================================================================
using System;
using System.Drawing;
using System.Collections;
using System.ComponentModel;
using System.Windows.Forms;
using System.Data;
using System.Diagnostics;

using MccDaq;

namespace ULTI01
{
	public class frmDataDisplay : System.Windows.Forms.Form
	{
		// Required by the Windows Form Designer
		private System.ComponentModel.IContainer components;

		public ToolTip ToolTip1;
		public Button cmdStopConvert;
		public Button cmdStartConvert;
		public TextBox txtExpChan;
		public Timer tmrConvert;
		public Label lblShowData;
		public Label lblValueRead;
		public Label lblChanPrompt;
		public Label lblDemoFunction;

		private MccDaq.MccBoard DaqBoard;
		private int UsesEXPs;

		public frmDataDisplay()
		{   
			// This call is required by the Windows Form Designer.
			InitializeComponent();

			//  Initiate error handling
			//   activating error handling will trap errors like
			//   bad channel numbers and non-configured conditions.
			//   Parameters:
			//     MccDaq.ErrorReporting.PrintAll :all warnings and errors encountered will be printed
			//     MccDaq.ErrorHandling.StopAll   :if an error is encountered, the program will stop
			
			MccDaq.ErrorInfo ULStat = MccDaq.MccService.ErrHandling(MccDaq.ErrorReporting.PrintAll, MccDaq.ErrorHandling.StopAll);
			

			//  Create a new MccBoard object for Board 0
			DaqBoard = new MccDaq.MccBoard(0);

			//  Determine if the board uses EXP boards for temperature measurements
			UsesEXPs = 0;
			ULStat = DaqBoard.BoardConfig.GetUsesExps(out UsesEXPs);
		}

		// Form overrides dispose to clean up the component list.
		protected override void  Dispose(bool Disposing)
		{
			if (Disposing)
			{
				if (components != null)
				{
					components.Dispose();
				}
			}
			base.Dispose(Disposing);
		}

			       	    
		#region Windows Form Designer generated code
		/// <summary>
		/// Required method for Designer support - do not modify
		/// the contents of this method with the code editor.
		/// </summary>
	    
		private void InitializeComponent()
		{

			this.components = new System.ComponentModel.Container();
			this.ToolTip1 = new System.Windows.Forms.ToolTip(this.components);
			this.cmdStopConvert = new System.Windows.Forms.Button();
			this.cmdStartConvert = new System.Windows.Forms.Button();
			this.txtExpChan = new System.Windows.Forms.TextBox();
			this.tmrConvert = new System.Windows.Forms.Timer(this.components);
			this.lblShowData = new System.Windows.Forms.Label();
			this.lblValueRead = new System.Windows.Forms.Label();
			this.lblChanPrompt = new System.Windows.Forms.Label();
			this.lblDemoFunction = new System.Windows.Forms.Label();
			this.SuspendLayout();
			//
			// cmdStopConvert
			//
			this.cmdStopConvert.BackColor = System.Drawing.SystemColors.Control;
			this.cmdStopConvert.Cursor = System.Windows.Forms.Cursors.Default;
			this.cmdStopConvert.Font = new System.Drawing.Font("Arial", 8.0F, System.Drawing.FontStyle.Regular, System.Drawing.GraphicsUnit.Point, ((byte)(0)));
			this.cmdStopConvert.ForeColor = System.Drawing.SystemColors.ControlText;
			this.cmdStopConvert.Location = new System.Drawing.Point(288, 152);
			this.cmdStopConvert.Name = "cmdStopConvert";
			this.cmdStopConvert.RightToLeft = System.Windows.Forms.RightToLeft.No;
			this.cmdStopConvert.Size = new System.Drawing.Size(57, 25);
			this.cmdStopConvert.TabIndex = 6;
			this.cmdStopConvert.Text = "Quit";
			this.cmdStopConvert.Visible = false;
			this.cmdStopConvert.Click += new System.EventHandler(this.cmdStopConvert_Click);
			//
			// cmdStartConvert
			//
			this.cmdStartConvert.BackColor = System.Drawing.SystemColors.Control;
			this.cmdStartConvert.Cursor = System.Windows.Forms.Cursors.Default;
			this.cmdStartConvert.Font = new System.Drawing.Font("Arial", 8.0F, System.Drawing.FontStyle.Regular, System.Drawing.GraphicsUnit.Point, ((byte)(0)));
			this.cmdStartConvert.ForeColor = System.Drawing.SystemColors.ControlText;
			this.cmdStartConvert.Location = new System.Drawing.Point(288, 152);
			this.cmdStartConvert.Name = "cmdStartConvert";
			this.cmdStartConvert.RightToLeft = System.Windows.Forms.RightToLeft.No;
			this.cmdStartConvert.Size = new System.Drawing.Size(57, 25);
			this.cmdStartConvert.TabIndex = 5;
			this.cmdStartConvert.Text = "Start";
			this.cmdStartConvert.Click += new System.EventHandler(this.cmdStartConvert_Click);
			//
			// txtExpChan
			//
			this.txtExpChan.AcceptsReturn = true;
			this.txtExpChan.AutoSize = false;
			this.txtExpChan.BackColor = System.Drawing.SystemColors.Window;
			this.txtExpChan.BorderStyle = System.Windows.Forms.BorderStyle.FixedSingle;
			this.txtExpChan.Cursor = System.Windows.Forms.Cursors.IBeam;
			this.txtExpChan.Font = new System.Drawing.Font("Arial", 8.0F, System.Drawing.FontStyle.Regular, System.Drawing.GraphicsUnit.Point, ((byte)(0)));
			this.txtExpChan.ForeColor = System.Drawing.SystemColors.WindowText;
			this.txtExpChan.Location = new System.Drawing.Point(312, 56);
			this.txtExpChan.MaxLength = 0;
			this.txtExpChan.Name = "txtExpChan";
			this.txtExpChan.RightToLeft = System.Windows.Forms.RightToLeft.No;
			this.txtExpChan.Size = new System.Drawing.Size(33, 25);
			this.txtExpChan.TabIndex = 0;
			this.txtExpChan.Text = "0";
			//
			// tmrConvert
			//
			this.tmrConvert.Interval = 250;
			this.tmrConvert.Tick += new System.EventHandler(this.tmrConvert_Tick);
			//
			// lblShowData
			//
			this.lblShowData.BackColor = System.Drawing.SystemColors.Window;
			this.lblShowData.Cursor = System.Windows.Forms.Cursors.Default;
			this.lblShowData.Font = new System.Drawing.Font("Arial", 8.0F, System.Drawing.FontStyle.Regular, System.Drawing.GraphicsUnit.Point, ((byte)(0)));
			this.lblShowData.ForeColor = System.Drawing.Color.Blue;
			this.lblShowData.Location = new System.Drawing.Point(280, 104);
			this.lblShowData.Name = "lblShowData";
			this.lblShowData.RightToLeft = System.Windows.Forms.RightToLeft.No;
			this.lblShowData.Size = new System.Drawing.Size(65, 17);
			this.lblShowData.TabIndex = 4;
			//
			// lblValueRead
			//
			this.lblValueRead.BackColor = System.Drawing.SystemColors.Window;
			this.lblValueRead.Cursor = System.Windows.Forms.Cursors.Default;
			this.lblValueRead.Font = new System.Drawing.Font("Arial", 8.0F, System.Drawing.FontStyle.Regular, System.Drawing.GraphicsUnit.Point, ((byte)(0)));
			this.lblValueRead.ForeColor = System.Drawing.SystemColors.WindowText;
			this.lblValueRead.Location = new System.Drawing.Point(24, 104);
			this.lblValueRead.Name = "lblValueRead";
			this.lblValueRead.RightToLeft = System.Windows.Forms.RightToLeft.No;
			this.lblValueRead.Size = new System.Drawing.Size(241, 17);
			this.lblValueRead.TabIndex = 3;
			this.lblValueRead.Text = "Temperature read from selected channel:";
			//
			// lblChanPrompt
			//
			this.lblChanPrompt.BackColor = System.Drawing.SystemColors.Window;
			this.lblChanPrompt.Cursor = System.Windows.Forms.Cursors.Default;
			this.lblChanPrompt.Font = new System.Drawing.Font("Arial", 8.0F, System.Drawing.FontStyle.Regular, System.Drawing.GraphicsUnit.Point, ((byte)(0)));
			this.lblChanPrompt.ForeColor = System.Drawing.SystemColors.WindowText;
			this.lblChanPrompt.Location = new System.Drawing.Point(8, 64);
			this.lblChanPrompt.Name = "lblChanPrompt";
			this.lblChanPrompt.RightToLeft = System.Windows.Forms.RightToLeft.No;
			this.lblChanPrompt.Size = new System.Drawing.Size(297, 17);
			this.lblChanPrompt.TabIndex = 1;
			this.lblChanPrompt.Text = "Enter the multiplexor channel to display (0 to 15): ";
			this.lblChanPrompt.TextAlign = System.Drawing.ContentAlignment.TopRight;
			//
			// lblDemoFunction
			//
			this.lblDemoFunction.BackColor = System.Drawing.SystemColors.Window;
			this.lblDemoFunction.Cursor = System.Windows.Forms.Cursors.Default;
			this.lblDemoFunction.Font = new System.Drawing.Font("Arial", 8.0F, System.Drawing.FontStyle.Regular, System.Drawing.GraphicsUnit.Point, ((byte)(0)));
			this.lblDemoFunction.ForeColor = System.Drawing.SystemColors.WindowText;
			this.lblDemoFunction.Location = new System.Drawing.Point(40, 16);
			this.lblDemoFunction.Name = "lblDemoFunction";
			this.lblDemoFunction.RightToLeft = System.Windows.Forms.RightToLeft.No;
			this.lblDemoFunction.Size = new System.Drawing.Size(273, 25);
			this.lblDemoFunction.TabIndex = 2;
			this.lblDemoFunction.Text = "Demonstration of Mccdaq.MccBoard.TIn()";
			this.lblDemoFunction.TextAlign = System.Drawing.ContentAlignment.TopCenter;
			//
			// frmDataDisplay
			//
			this.AcceptButton = this.cmdStartConvert;
			this.AutoScaleBaseSize = new System.Drawing.Size(6, 13);
			this.BackColor = System.Drawing.SystemColors.Window;
			this.ClientSize = new System.Drawing.Size(360, 190);
			this.Controls.AddRange(new System.Windows.Forms.Control[] {this.cmdStopConvert, this.cmdStartConvert, this.txtExpChan, this.lblShowData, this.lblValueRead, this.lblChanPrompt, this.lblDemoFunction});
			this.Font = new System.Drawing.Font("Arial", 8.25F, System.Drawing.FontStyle.Bold, System.Drawing.GraphicsUnit.Point, ((byte)(0)));
			this.ForeColor = System.Drawing.SystemColors.WindowText;
			this.Location = new System.Drawing.Point(7, 103);
			this.Name = "frmDataDisplay";
			this.StartPosition = System.Windows.Forms.FormStartPosition.Manual;
			this.Text = "Universal Library Temperature Measurement";
			this.ResumeLayout(false);

		}

	#endregion

		/// <summary>
		/// The main entry point for the application.
		/// </summary>
		[STAThread]
		static void Main() 
		{
			Application.Run(new frmDataDisplay());
		}
	    	

		private void cmdStartConvert_Click(object eventSender, System.EventArgs eventArgs) /* Handles cmdStartConvert.Click */
		{
			if (int.Parse(txtExpChan.Text) < 0) 
				txtExpChan.Text = "0";

			if (int.Parse(txtExpChan.Text) > 15) 
				txtExpChan.Text = "15";

			cmdStartConvert.Visible = false;
			cmdStopConvert.Visible = true;
			tmrConvert.Enabled = true;
		}


		private void cmdStopConvert_Click(object eventSender, System.EventArgs eventArgs) /* Handles cmdStopConvert.Click */
		{
			Application.Exit();
		}


		private void tmrConvert_Tick(object eventSender, System.EventArgs eventArgs) /* Handles tmrConvert.Tick */
		{
			tmrConvert.Stop();

			//  Collect the data with Mccdaq.MccBoard.TIn()

			//  Parameters:
			//    Chan       :the A/D and channel number; starts at 16
			//                 calculated by (ADChan + 1) * 16 + EXPChan%
			//    MccScale    :the temperature scale (F, C or K)
			//    DataValue  :the name for the value collected
			//    ADChan     :usually channel 0 for CIO-EXP16
			short ADChan = 0; //  allows access to 16 channels on the EXP board

			//  increasing this number allows access to upper
			//  bank on EXP32 and additional EXP boards
			MccDaq.TempScale MccScale = MccDaq.TempScale.Celsius;
			int Chan = 0;
			if (UsesEXPs > 0)
				Chan = int.Parse(txtExpChan.Text) + (ADChan + 1) * 16;
			else
				Chan = int.Parse(txtExpChan.Text);

			float TempValue = 0.0f;
			MccDaq.ThermocoupleOptions Options = MccDaq.ThermocoupleOptions.Filter;
			MccDaq.ErrorInfo ULStat = DaqBoard.TIn( Chan, MccScale, out TempValue, Options);

			if (ULStat.Value == MccDaq.ErrorInfo.ErrorCode.NoErrors)
				tmrConvert.Start();

			lblShowData.Text = TempValue.ToString("0") + "°C"; //  print the value    
		}


		private void txtExpChan_TextChanged(object eventSender, System.EventArgs eventArgs) /* Handles txtExpChan.TextChanged */
		{
			tmrConvert.Enabled = false;
			cmdStartConvert.Visible = true;
			cmdStopConvert.Visible = false;
		}

	}
}

File: measurecompute/DAQ/VB.NET/ULTI01.NET/ULTI01.vb
'==============================================================================

' File:                         ULTI01.VB

' Library Call Demonstrated:    Mccdaq.MccBoard.TIn()

' Purpose:                      Reads a multiplexer input channel.

' Demonstration:                Displays the temperature input on a MUX
'                               channel.

' Other Library Calls:          MccDaq.MccService.ErrHandling()

' Special Requirements:         Unless the board at BoardNum(=0) does not use
'                               EXP boards for temperature measurements(the
'                               CIO-DAS-TC,for example), it must
'                               have an A/D converter with an attached EXP
'                               board.  Thermocouples must be wired to EXP
'                              channels selected.

'==============================================================================
Option Strict Off
Option Explicit On 
Public Class frmDataDisplay
    Inherits System.Windows.Forms.Form


    ' Create a new MccBoard object for Board 0
    Private DaqBoard As MccDaq.MccBoard = New MccDaq.MccBoard(0)
    Dim UsesEXPs As Integer

    Private Sub cmdStartConvert_Click(ByVal eventSender As System.Object, ByVal eventArgs As System.EventArgs) Handles cmdStartConvert.Click

        If Val(txtExpChan.Text) < 0 Then txtExpChan.Text = "0"
        If Val(txtExpChan.Text) > 15 Then txtExpChan.Text = "15"

        cmdStartConvert.Visible = False
        cmdStopConvert.Visible = True
        tmrConvert.Enabled = True

    End Sub

    Private Sub cmdStopConvert_Click(ByVal eventSender As System.Object, ByVal eventArgs As System.EventArgs) Handles cmdStopConvert.Click

        End

    End Sub

    Private Sub tmrConvert_Tick(ByVal eventSender As System.Object, ByVal eventArgs As System.EventArgs) Handles tmrConvert.Tick
        Dim ULStat As MccDaq.ErrorInfo
        Dim TempValue As Single
        Dim Options As MccDaq.ThermocoupleOptions
        Dim Chan As Integer
        Dim MccScale As MccDaq.TempScale
        Dim ADChan As Short

        ' Collect the data with Mccdaq.MccBoard.TIn()

        ' Parameters:
        '   Chan       :the A/D and channel number; starts at 16
        '                calculated by (ADChan + 1) * 16 + EXPChan%
        '   MccScale    :the temperature scale (F, C or K)
        '   DataValue  :the name for the value collected
        '   ADChan     :usually channel 0 for CIO-EXP16

        ADChan = 0 ' allows access to 16 channels on the EXP board
        ' increasing this number allows access to upper
        ' bank on EXP32 and additional EXP boards

        MccScale = MccDaq.TempScale.Celsius
        If UsesEXPs > 0 Then
            Chan = Integer.Parse(txtExpChan.Text) + (ADChan + 1) * 16
        Else
            Chan = Integer.Parse(txtExpChan.Text)
        End If

        lblShowData.Text = TempValue.ToString("0") + "C" ' print the value
        Options = MccDaq.ThermocoupleOptions.Filter

        tmrConvert.Stop()

        ULStat = DaqBoard.TIn(Chan, MccScale, TempValue, Options)

        If ULStat.Value <> MccDaq.ErrorInfo.ErrorCode.NoErrors Then Stop

        lblShowData.Text = TempValue.ToString("0") + "°C" ' print the value
        
        tmrConvert.Start()

    End Sub

    Private Sub txtExpChan_TextChanged(ByVal eventSender As System.Object, ByVal eventArgs As System.EventArgs) Handles txtExpChan.TextChanged

        tmrConvert.Enabled = False
        cmdStartConvert.Visible = True
        cmdStopConvert.Visible = False


    End Sub

#Region "Windows Form Designer generated code "
    Public Sub New()
        MyBase.New()

        'This call is required by the Windows Form Designer.
        InitializeComponent()

        InitUL()

    End Sub
    'Form overrides dispose to clean up the component list.
    Protected Overloads Overrides Sub Dispose(ByVal Disposing As Boolean)
        If Disposing Then
            If Not components Is Nothing Then
                components.Dispose()
            End If
        End If
        MyBase.Dispose(Disposing)
    End Sub
    'Required by the Windows Form Designer
    Private components As System.ComponentModel.IContainer
    Public ToolTip1 As System.Windows.Forms.ToolTip
    Public WithEvents cmdStopConvert As System.Windows.Forms.Button
    Public WithEvents cmdStartConvert As System.Windows.Forms.Button
    Public WithEvents txtExpChan As System.Windows.Forms.TextBox
    Public WithEvents tmrConvert As System.Windows.Forms.Timer
    Public WithEvents lblShowData As System.Windows.Forms.Label
    Public WithEvents lblValueRead As System.Windows.Forms.Label
    Public WithEvents lblChanPrompt As System.Windows.Forms.Label
    Public WithEvents lblDemoFunction As System.Windows.Forms.Label
    'NOTE: The following procedure is required by the Windows Form Designer
    'It can be modified using the Windows Form Designer.
    'Do not modify it using the code editor.
    <System.Diagnostics.DebuggerStepThrough()> Private Sub InitializeComponent()
        Me.components = New System.ComponentModel.Container()
        Me.ToolTip1 = New System.Windows.Forms.ToolTip(Me.components)
        Me.cmdStopConvert = New System.Windows.Forms.Button()
        Me.cmdStartConvert = New System.Windows.Forms.Button()
        Me.txtExpChan = New System.Windows.Forms.TextBox()
        Me.tmrConvert = New System.Windows.Forms.Timer(Me.components)
        Me.lblShowData = New System.Windows.Forms.Label()
        Me.lblValueRead = New System.Windows.Forms.Label()
        Me.lblChanPrompt = New System.Windows.Forms.Label()
        Me.lblDemoFunction = New System.Windows.Forms.Label()
        Me.SuspendLayout()
        '
        'cmdStopConvert
        '
        Me.cmdStopConvert.BackColor = System.Drawing.SystemColors.Control
        Me.cmdStopConvert.Cursor = System.Windows.Forms.Cursors.Default
        Me.cmdStopConvert.Font = New System.Drawing.Font("Arial", 8.0!, System.Drawing.FontStyle.Regular, System.Drawing.GraphicsUnit.Point, CType(0, Byte))
        Me.cmdStopConvert.ForeColor = System.Drawing.SystemColors.ControlText
        Me.cmdStopConvert.Location = New System.Drawing.Point(288, 152)
        Me.cmdStopConvert.Name = "cmdStopConvert"
        Me.cmdStopConvert.RightToLeft = System.Windows.Forms.RightToLeft.No
        Me.cmdStopConvert.Size = New System.Drawing.Size(57, 25)
        Me.cmdStopConvert.TabIndex = 6
        Me.cmdStopConvert.Text = "Quit"
        Me.cmdStopConvert.Visible = False
        '
        'cmdStartConvert
        '
        Me.cmdStartConvert.BackColor = System.Drawing.SystemColors.Control
        Me.cmdStartConvert.Cursor = System.Windows.Forms.Cursors.Default
        Me.cmdStartConvert.Font = New System.Drawing.Font("Arial", 8.0!, System.Drawing.FontStyle.Regular, System.Drawing.GraphicsUnit.Point, CType(0, Byte))
        Me.cmdStartConvert.ForeColor = System.Drawing.SystemColors.ControlText
        Me.cmdStartConvert.Location = New System.Drawing.Point(288, 152)
        Me.cmdStartConvert.Name = "cmdStartConvert"
        Me.cmdStartConvert.RightToLeft = System.Windows.Forms.RightToLeft.No
        Me.cmdStartConvert.Size = New System.Drawing.Size(57, 25)
        Me.cmdStartConvert.TabIndex = 5
        Me.cmdStartConvert.Text = "Start"
        '
        'txtExpChan
        '
        Me.txtExpChan.AcceptsReturn = True
        Me.txtExpChan.AutoSize = False
        Me.txtExpChan.BackColor = System.Drawing.SystemColors.Window
        Me.txtExpChan.BorderStyle = System.Windows.Forms.BorderStyle.FixedSingle
        Me.txtExpChan.Cursor = System.Windows.Forms.Cursors.IBeam
        Me.txtExpChan.Font = New System.Drawing.Font("Arial", 8.0!, System.Drawing.FontStyle.Regular, System.Drawing.GraphicsUnit.Point, CType(0, Byte))
        Me.txtExpChan.ForeColor = System.Drawing.SystemColors.WindowText
        Me.txtExpChan.Location = New System.Drawing.Point(312, 56)
        Me.txtExpChan.MaxLength = 0
        Me.txtExpChan.Name = "txtExpChan"
        Me.txtExpChan.RightToLeft = System.Windows.Forms.RightToLeft.No
        Me.txtExpChan.Size = New System.Drawing.Size(33, 25)
        Me.txtExpChan.TabIndex = 0
        Me.txtExpChan.Text = "0"
        '
        'tmrConvert
        '
        Me.tmrConvert.Interval = 250
        '
        'lblShowData
        '
        Me.lblShowData.BackColor = System.Drawing.SystemColors.Window
        Me.lblShowData.Cursor = System.Windows.Forms.Cursors.Default
        Me.lblShowData.Font = New System.Drawing.Font("Arial", 8.0!, System.Drawing.FontStyle.Regular, System.Drawing.GraphicsUnit.Point, CType(0, Byte))
        Me.lblShowData.ForeColor = System.Drawing.Color.Blue
        Me.lblShowData.Location = New System.Drawing.Point(280, 104)
        Me.lblShowData.Name = "lblShowData"
        Me.lblShowData.RightToLeft = System.Windows.Forms.RightToLeft.No
        Me.lblShowData.Size = New System.Drawing.Size(65, 17)
        Me.lblShowData.TabIndex = 4
        '
        'lblValueRead
        '
        Me.lblValueRead.BackColor = System.Drawing.SystemColors.Window
        Me.lblValueRead.Cursor = System.Windows.Forms.Cursors.Default
        Me.lblValueRead.Font = New System.Drawing.Font("Arial", 8.0!, System.Drawing.FontStyle.Regular, System.Drawing.GraphicsUnit.Point, CType(0, Byte))
        Me.lblValueRead.ForeColor = System.Drawing.SystemColors.WindowText
        Me.lblValueRead.Location = New System.Drawing.Point(24, 104)
        Me.lblValueRead.Name = "lblValueRead"
        Me.lblValueRead.RightToLeft = System.Windows.Forms.RightToLeft.No
        Me.lblValueRead.Size = New System.Drawing.Size(241, 17)
        Me.lblValueRead.TabIndex = 3
        Me.lblValueRead.Text = "Temperature read from selected channel:"
        '
        'lblChanPrompt
        '
        Me.lblChanPrompt.BackColor = System.Drawing.SystemColors.Window
        Me.lblChanPrompt.Cursor = System.Windows.Forms.Cursors.Default
        Me.lblChanPrompt.Font = New System.Drawing.Font("Arial", 8.0!, System.Drawing.FontStyle.Regular, System.Drawing.GraphicsUnit.Point, CType(0, Byte))
        Me.lblChanPrompt.ForeColor = System.Drawing.SystemColors.WindowText
        Me.lblChanPrompt.Location = New System.Drawing.Point(8, 64)
        Me.lblChanPrompt.Name = "lblChanPrompt"
        Me.lblChanPrompt.RightToLeft = System.Windows.Forms.RightToLeft.No
        Me.lblChanPrompt.Size = New System.Drawing.Size(297, 17)
        Me.lblChanPrompt.TabIndex = 1
        Me.lblChanPrompt.Text = "Enter the multiplexor channel to display (0 to 15): "
        Me.lblChanPrompt.TextAlign = System.Drawing.ContentAlignment.TopRight
        '
        'lblDemoFunction
        '
        Me.lblDemoFunction.BackColor = System.Drawing.SystemColors.Window
        Me.lblDemoFunction.Cursor = System.Windows.Forms.Cursors.Default
        Me.lblDemoFunction.Font = New System.Drawing.Font("Arial", 8.0!, System.Drawing.FontStyle.Regular, System.Drawing.GraphicsUnit.Point, CType(0, Byte))
        Me.lblDemoFunction.ForeColor = System.Drawing.SystemColors.WindowText
        Me.lblDemoFunction.Location = New System.Drawing.Point(40, 16)
        Me.lblDemoFunction.Name = "lblDemoFunction"
        Me.lblDemoFunction.RightToLeft = System.Windows.Forms.RightToLeft.No
        Me.lblDemoFunction.Size = New System.Drawing.Size(273, 25)
        Me.lblDemoFunction.TabIndex = 2
        Me.lblDemoFunction.Text = "Demonstration of Mccdaq.MccBoard.TIn()"
        Me.lblDemoFunction.TextAlign = System.Drawing.ContentAlignment.TopCenter
        '
        'frmDataDisplay
        '
        Me.AcceptButton = Me.cmdStartConvert
        Me.AutoScaleBaseSize = New System.Drawing.Size(6, 13)
        Me.BackColor = System.Drawing.SystemColors.Window
        Me.ClientSize = New System.Drawing.Size(360, 190)
        Me.Controls.AddRange(New System.Windows.Forms.Control() {Me.cmdStopConvert, Me.cmdStartConvert, Me.txtExpChan, Me.lblShowData, Me.lblValueRead, Me.lblChanPrompt, Me.lblDemoFunction})
        Me.Font = New System.Drawing.Font("Arial", 8.25!, System.Drawing.FontStyle.Bold, System.Drawing.GraphicsUnit.Point, CType(0, Byte))
        Me.ForeColor = System.Drawing.SystemColors.WindowText
        Me.Location = New System.Drawing.Point(7, 103)
        Me.Name = "frmDataDisplay"
        Me.StartPosition = System.Windows.Forms.FormStartPosition.Manual
        Me.Text = "Universal Library Temperature Measurement"
        Me.ResumeLayout(False)

    End Sub
#End Region

#Region "Universal Library Initialization - Expand this region to change error handling, etc."

    Private Sub InitUL()

        Dim ConfigItem As Short
        Dim DevNum As Short
        Dim InfoType As Short
        Dim ULStat As MccDaq.ErrorInfo

        ' declare revision level of Universal Library

        ULStat = MccDaq.MccService.DeclareRevision(MccDaq.MccService.CurrentRevNum)

        ' Initiate error handling
        '  activating error handling will trap errors like
        '  bad channel numbers and non-configured conditions.
        '  Parameters:
        '    MccDaq.ErrorReporting.PrintAll :all warnings and errors encountered will be printed
        '    MccDaq.ErrorHandling.StopAll   :if any error is encountered, the program will stop


        ULStat = MccDaq.MccService.ErrHandling(MccDaq.ErrorReporting.PrintAll, MccDaq.ErrorHandling.StopAll)
        If ULStat.Value <> MccDaq.ErrorInfo.ErrorCode.NoErrors Then
            Stop
        End If

        ' Determine if the board uses EXP boards for temperature measurements
        DevNum = 0
        UsesEXPs = 0
        ULStat = DaqBoard.BoardConfig.GetUsesExps(UsesEXPs)
        If (ULStat.Value <> MccDaq.ErrorInfo.ErrorCode.NoErrors) Then UsesEXPs = 0

    End Sub
#End Region

End Class